src/manifolds/ConnectionManifold.jl

```diff
@@ -0,0 +1,404 @@
+    M::AbstractDecoratorManifold,
+    p,
+    B::AbstractBasis;
+    kwargs...,
+)
+
+@doc raw"""
+    christoffel_symbols_second_jacobian(
+        M::AbstractManifold,
+        p,
+        B::AbstractBasis;
+        backend::AbstractDiffBackend = diff_backend(),
+    )
+
+Get partial derivatives of the Christoffel symbols of the second kind
+for manifold `M` at `p` with respect to the coordinates of `B`, i.e.
+
+```math
+\frac{∂}{∂ p^l} Γ^{k}_{ij} = Γ^{k}_{ij,l}.
+```
+
+The dimensions of the resulting multi-dimensional array are ordered ``(i,j,k,l)``.
+"""
+christoffel_symbols_second_jacobian(::AbstractManifold, ::Any, B::AbstractBasis)
+function christoffel_symbols_second_jacobian(
+    M::AbstractManifold,
+    p,
+    B::AbstractBasis;
+    backend::AbstractDiffBackend=diff_backend(),
+)
+    n = size(p, 1)
+    ∂Γ = reshape(
+        _jacobian(q -> christoffel_symbols_second(M, q, B; backend=backend), p, backend),
+        n,
+        n,
+        n,
+        n,
+    )
+    return ∂Γ
+end
+@decorator_transparent_signature christoffel_symbols_second_jacobian(
+    M::AbstractDecoratorManifold,
+    p,
+    B::AbstractBasis;
+    kwargs...,
+)
+
+connection(M::ConnectionManifold) = M.connection
+
+Base.copyto!(M::AbstractConnectionManifold, q, p) = copyto!(M.manifold, q, p)
+Base.copyto!(M::AbstractConnectionManifold, Y, p, X) = copyto!(M.manifold, Y, p, X)
+
+@doc raw"""
+    exp(M::AbstractConnectionManifold, p, X)
+
+Copute the exponential map on the [`AbstractConnectionManifold`](@ref) `M` equipped with
+corresponding affine metric.
+
+If the metric was declared the default metric using [`is_default_metric`](@ref), this method
+falls back to `exp(M, p, X)`.
+
+Otherwise it numerically integrates the underlying ODE, see [`solve_exp_ode`](@ref).
+Currently, the numerical integration is only accurate when using a single
+coordinate chart that covers the entire manifold. This excludes coordinates
+in an embedded space.
+"""
+exp(::AbstractConnectionManifold, ::Any...)
+
+@decorator_transparent_fallback function exp!(M::AbstractConnectionManifold, q, p, X)
+    tspan = (0.0, 1.0)
+    A = get_default_atlas(M)
+    i = get_chart_index(M, A, p)
+    B = induced_basis(M, A, i, TangentSpace)
+    sol = solve_exp_ode(M, p, X, tspan, B; dense=false, saveat=[1.0])
+    n = length(p)
+    return copyto!(q, sol.u[1][(n + 1):end])
+end
+
+"""
+    gaussian_curvature(M::AbstractManifold, p, B::AbstractBasis; backend::AbstractDiffBackend = diff_backend())
+
+Compute the Gaussian curvature of the manifold `M` at the point `p` using basis `B`.
+"""
+gaussian_curvature(::AbstractManifold, ::Any, ::AbstractBasis)
+function gaussian_curvature(M::AbstractManifold, p, B::AbstractBasis; kwargs...)
+    return ricci_curvature(M, p, B; kwargs...) / 2
+end
+@decorator_transparent_signature gaussian_curvature(
+    M::AbstractDecoratorManifold,
+    p,
+    B::AbstractBasis;
+    kwargs...,
+)
+
+function injectivity_radius(M::AbstractConnectionManifold, p)
```

Comment:
Could this instead be done with transparency?

Reply:
Actually I don't think this is even correct, I just copied it from the metric manifold. I'd prefer to resolve this in a separate PR.

Reply:
Ok, that's also fine with me.